Assets/Scripts/Effects/BlobShadowProjector.cs

```diff
@@ -0,0 +1,46 @@
+﻿using UnityEngine;
+
+namespace Effects
+{
+
+    public class BlobShadowProjector : MonoBehaviour
+    {
+        private Transform parent = null;
+
+        void Start()
+        {
+            parent = this.transform.parent;
+
+            TurnDynamicShadowsOnIfTrue(false);
+        }
+
+        private void OnDestroy()
+        {
+            TurnDynamicShadowsOnIfTrue(true);
+        }
+
+        private void TurnDynamicShadowsOnIfTrue(bool b)
+        {
+            if (b)
+            {
+                foreach (Transform child in parent)
+                {
+                    if (child.GetComponent<Renderer>() != null)
+                    {
+                        child.GetComponent<Renderer>().shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.On;
+                    }
+                }
+            }
+            else
+            {
+                foreach (Transform child in parent)
+                {
+                    if (child.GetComponent<Renderer>() != null)
+                    {
+                        child.GetComponent<Renderer>().shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.Off;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
```C#
private void SetDynamicShadowsOnParent(bool enabled)
{
    foreach (Transform child in parent)
    {
        var renderer = child.GetComponent<Renderer>();
        if (renderer)
        {
            renderer.shadowCastingMode = enabled ? ShadowCastingMode.On : ShadowCastingMode.Off;
        }
    }
}
```

Reply:
It looks nice. I was trying to avoid checking the boolean separately for every children. If it's fine, then I'll gladly paste this.

Reply:
If you want to avoid the boolean check you can do:
```C#
var mode = enabled ? ShadowCastingMode.On : ShadowCastingMode.Off;
foreach (Transform child in parent)
....
        renderer.shadowCastingMode = mode;
....
```

Reply:
Also creating that new variable for every children to store the renderer bugs me a bit. If it's fine then I'll add it.

Reply:
Variable creation has 0 cost. The variable is created because `GetComponent` has a slight performance impact, so is the better of the options.

Reply:
Ok, nice. Thanks for the advice. I'll add your suggestions.